packages/api/internal/orchestrator/create_instance.go

```diff
@@ -105,6 +105,20 @@ func (o *Orchestrator) CreateSandbox(
 
 	telemetry.ReportEvent(childCtx, "Got FC version info")
 
+	var sbxDomain *string
+	if team.Team.ClusterID != nil {
+		cluster, ok := o.clusters.GetClusterById(*team.Team.ClusterID)
+		if !ok {
+			return nil, &api.APIError{
+				Code:      http.StatusInternalServerError,
+				ClientMsg: "Error while looking for sandbox cluster information",
+				Err:       fmt.Errorf("cannot access cluster %s associated with team id %s that spawned sandbox %s", *team.Team.ClusterID, team.Team.ID, sandboxID),
+			}
+		}
+
+		sbxDomain = cluster.SandboxDomain
+	}
+
```

Comment:
Moved here from sandbox create handler, because we are now just re-using the sbx api response in this handler and not creating it again from received values.

https://github.com/e2b-dev/infra/pull/872